core/signeddata.go

```diff
@@ -34,6 +34,7 @@ var (
 	_ SignedData = Signature{}
 	_ SignedData = SignedVoluntaryExit{}
 	_ SignedData = VersionedSignedBlindedBeaconBlock{}
+	_ SignedData = VersionedSignedValidatorRegistration{} // name in the builder spec SignedValidatorRegistrationV1 or SignedBuilderRegistration
```

Comment:
not sure on preferred naming for this section 

validatorRegistration or BuilderRegistration 

within go-eth2-client it follows the builder spec name validatorRegistration

Reply:
aligning with `eth2api.VersionedSignedValidatorRegistration` is good I think